scipy/optimize/zeros.c

```diff
@@ -129,11 +129,12 @@ call_solver(solver_type solver, PyObject *self, PyObject *args)
                         "Failed to converge after %d iterations.",
                         solver_stats.iterations);
                 PyErr_SetString(PyExc_RuntimeError, msg);
-                flag = 1;
                 return NULL;
             }
+            flag = CONVERR;
```

Comment:
Approve of the use of the #define values `CONVERGED`, `CONVERR` etc.
It looks as though the value of `flag` is set on line 134 to `CONVERR` but is overridden just below on line 137 before this value has a chance to be used?

Reply:
* moved line 137 into `else{}`, _ie_: if not converged then must be converged
* also added test to check this works